tests/test_broadcast.py

```diff
@@ -248,6 +249,29 @@ async def test_broadcast_filter() -> None:
     assert (await receiver.receive()) == 15
 
 
+async def test_broadcast_filter_type_guard() -> None:
+    """Ensure filter type guard works."""
+    chan = Broadcast[int | str](name="input-chan")
+    sender = chan.new_sender()
+
+    def _is_int(num: int | str) -> TypeGuard[int]:
+        return isinstance(num, int)
+
+    # filter out all numbers less than 10.
+    receiver = chan.new_receiver().filter(_is_int)
+
+    await sender.send("hello")
+    await sender.send(8)
+
+    message = await receiver.receive()
+    assert message == 8
+    match message:
+        case int():
+            assert message == 8
```

Comment:
Why is it necessary to have this check twice? (in line 267 and 270 ? ` assert message == 8` )

Reply:
I guess it could just be a `pass` inside the `case`, true. I just needed to put something :D

Reply:
I just changed the test to make sure the `case int()` is executed. `mypy` should verify this too thannks to the `assert_never`, but just in case.

Reply:
Thanks! Now it is clear for me why it was necessary :)